modules/javafx.controls/src/main/java/javafx/scene/chart/CategoryAxis.java

```diff
@@ -369,7 +370,15 @@ private double calculateNewFirstPos(double length, double catSpacing) {
                 if (side.isVertical() && tickLabelRotation != 0) {
                     tickLabelRotation = 0;
                 }
+            } else {
+                if (side.isHorizontal() && tickLabelRotation == 90) {
+                    tickLabelRotation = 0;
+                }
+                if (side.isVertical() && tickLabelRotation == 0) {
+                    tickLabelRotation = 90;
+                }
             }
+            if (!isAutoRanging()) setTickLabelRotation(tickLabelRotation);
```

Comment:
At first, I thought this setTickLabelRotation() call violates the method javadoc above. On a second look, I found that the javadoc is applicable only if AutoRanging() is true.
Method calls - calculateNewSpacing() and calculateNewFirstPos() also set properties if AutoRanging is false. Hence, this fix seems OK to me.

Reply:
> 
> 
> At first, I thought this setTickLabelRotation() call violates the method javadoc above. On a second look, I found that the javadoc is applicable only if AutoRanging() is true.
> Method calls - calculateNewSpacing() and calculateNewFirstPos() also set properties if AutoRanging is false. Hence, this fix seems OK to me.

hmm ... indeed they do, unexpected for me, reading the must-not-effect-the-state in the javadoc ;) Sounds like a doc error, if the intention was to clearly state that the method shouldn't do the auto-ranging itself?